### ISSUE TITLE
Only post touched fields

### ISSUE THREAD

**Issue by cjcheshire** [jaredpalmer/formik#1390]
## ❓Question

Hey, does anyone have an example of how to only submit touched values. I'm currently using the formik HOC but the handleSubmit function doesn't have access to the touched prop.

**Comment by Funkyskank** [comment]
If you are using withFormik HOC, it's `handleSubmit` have second argument with formikBag and props passed to component, including `touched`:
```javascript
const formikEnhancer = withFormik({
  handleSubmit: (payload, bag: { props: { touched } }) => {
     console.log(touched);
  },
  displayName: "MyForm",
});
```
But all fields (that mentioned in initialValues) will be marked as touched on submit. So it will not help.

**Comment by cjcheshire** [comment]
@funkyskank thanks, can’t believe I missed that. 

It’s certainly a workaround but I could set up the equivalent of a hidden field in web land and populate via the fields that get touched.

**Comment by bke-daniel** [comment]
Agree, this would be interesting in regards of atomic (database) updates.

**Comment by bganicky** [comment]
Hi, I need a granular field update as well and can't figure out how to do that. I'm using `<Formik>` component and there's no `touched` on the `onSubmit`'s `formikBag` object.

Is there a specific reason why it's not present there? Just to be clear, documentation is quite honest about it but doesn't mention any particular reason:

> Note: errors, touched, status and all event handlers are NOT included in the FormikBag.

**Comment by piotrpawelzieba** [comment]
any update?

**Comment by Kepro** [comment]
looks like we need to create a deep diff between old and new object...

**Comment by johnrom** [comment]
v3 will provide access to get Formik's state using `getState()`, which means you would be able to traverse getState().touched

**Comment by talyh** [comment]
@johnrom  - is there a planned timeline for v3 and the ability to perceive `touched` inside `onSubmit`?

**Comment by zeyadetman** [comment]
As touched fields are not returned inside `onSubmit` I have a workaround to get them inside your component.

```jsx
const [touchedFields, setTouchedFields] = useState<Set<string>>(new Set()); // this will be the same as formik touched

<Field
  name="firstName"
  onChangeText={(value) => {
    setTouchedFields(
      (prevState) => new Set(prevState.add('firstName'))
    );
    handleChange('firstName').apply(null, [value]);
  }}
  component={Input}
/>  
```

**Comment by ThomasEsseul** [comment]
If anyone needs it, here's my workaround with the <Formik> component : 
Instead of doing
`
const mySubmit = (values: MyFields) => { ... }

return (
  <Formik onSubmit={mySubmit}>
    { props => 
      ... 
      <Button type="submit" />
    }
  </Formik>
)
`

I do : 
`
const mySubmit = (values: MyFields, touched: FormikTouched<MyFields>) => { ... }

return (
  <Formik onSubmit={ ( ) => { } }>
    { props => 
      ... 
      <Button type="button" onClick={ ( ) => mySubmit(props.values, props.touched) } />
    }
  </Formik>
)
`

Still, It would be great to have direct access through FormikBag (v3 🤞)

**Comment by dbgordon09** [comment]
I think it would be great if we could access the values for `touched` in the submission handler BEFORE Formik touches all fields.

It would be great if the pre-submission `touched` values were captured and then passed to `onSubmit` so that those values could be used for more bespoke submission logic like only returning modifed fields to a `PATCH`.

**Comment by Harsha8399** [comment]
I think it would be ok to use handleSubmit out of formik on submit.That would give access to all touched fields and can all other formik features would still work.
